### PR TITLE
TrajectoryCollator

### PR DESCRIPTION
**Pull request by gaschler**
Introduces TrajectorCollator, which collates sensor data ignoring
other trajectories.
Tests the same.

[RFC=0008](https://github.com/googlecartographer/rfcs/blob/master/text/0008-collator-interface.md)

### CODE REVIEW

File: cartographer/sensor/test_helpers.h [hunk 1]
```diff
@@ -29,6 +36,47 @@ MATCHER_P(Near, point, std::string(negation ? "Doesn't" : "Does") + " match.") {
   return arg.isApprox(point, 0.001f);
 }
 
+namespace test {
+
+typedef std::tuple<int, std::string, common::Time> CollatorOutput;
```
Comment (cschuet) [inline cartographer/sensor/test_helpers.h]: can you add /* trajectory_id */ and such to make clear what those ints and strings are?

File: cartographer/sensor/test_helpers.h [hunk 2]
```diff
@@ -29,6 +36,47 @@ MATCHER_P(Near, point, std::string(negation ? "Doesn't" : "Does") + " match.") {
...
+                                          std::string sensor_id, int time) {
+    return CollatorInput{
+        trajectory_id,
+        MakeDispatchable(sensor_id,
+                         OdometryData{common::FromUniversal(time),
+                                      transform::Rigid3d::Identity()})};
+  }
+  void MoveToCollator(CollatorInterface* collator) {
+    expected_output =
+        CollatorOutput(trajectory_id, data->GetSensorId(), data->GetTime());
```
Comment (cschuet) [inline cartographer/sensor/test_helpers.h]: It's slightly unusual that you set the output here, although you could have set it at construction time. Then you could even declare it const.

Comment (gaschler) [reply cartographer/sensor/test_helpers.h]: Done

File: cartographer/sensor/test_helpers.h (hunk 1 above)
Comment (gaschler) [reply cartographer/sensor/test_helpers.h]: Done

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 1.

**Pull request merged**